RPerformanceTracking/Private/_RPTTrackingManager.m

```
@@ -160,12 +160,12 @@ - (void)updateConfiguration
     NSString *thisVersion   = [thisBundle objectForInfoDictionaryKey:@"CFBundleShortVersionString"];
     
     NSBundle *appBundle     = NSBundle.mainBundle;
-    NSString *appIdentifier = appBundle.bundleIdentifier;
+    NSString *relayAppID    = [appBundle objectForInfoDictionaryKey:@"RPTRelayAppID"];
```

This key is something set by the developer in their app's plist so we should do the same non-zero length checks that we do for the subscription key.

---

Got it.. Ok

---

Then, If the length is zero, we do send without app id?

---

Incase if developer forgets to set the key in his app's plist will it throw any exception hinting the developer to look at his plist file to set key?

---

Android fails fast, because this config value is not gonna change in production apps - but it might lead to problems during integration (user forgets and gets frustrated when no data shows in the portal)
@donnie-jp what's your take on this?

---

@superbgv @NemoOudeis There's an assert that fires in debug builds if that key isn't set saying `Your application's Info.plist must contain a key 'RPTRelayAppID' set to...`